src/irc/core/irc-cap.c

```diff
@@ -79,73 +79,148 @@ static void cap_emit_signal (IRC_SERVER_REC *server, char *cmd, char *args)
 	g_free(signal_name);
 }
 
+static gboolean parse_cap_name(char *name, char **key, char **val)
+{
+	g_return_val_if_fail(name != NULL, FALSE);
+	g_return_val_if_fail(name[0] != '\0', FALSE);
+
+	const char *eq = strchr(name, '=');
+	/* KEY only value */
+	if (eq == NULL) {
+		*key = g_strdup(name);
+		*val = NULL;
+		return TRUE;
+	}
+	/* Some values are in a KEY=VALUE form, parse them */
+	else if (eq[1] != '\0') {
+		*key = g_strndup(name, (gsize)(eq - name));
+		*val = g_strdup(eq + 1);
+		return TRUE;
+	}
+	/* If the string ends after the '=' consider the value
+	 * as invalid */
+	else {
```

Comment:
does the spec not allow empty values?

Reply:
That's a good question, the spec states the format is `<name>[=<value>]` with no further indications.